### PR TITLE
Distinguish between public and private contact in contact actions

### LINKED ISSUE #10502: Unblock blocked accounts via  /contact/blocked doesn't work
Unblocking blocked accounts doesn't work in the users personal settings /contact/blocked  (single or mass unblocking)

Currently you have to open every contacts profile and unblock via pull down menu (Frio)

Node: Friendica 2021.07 stable / Git / PHP 7.4

### PR DESCRIPTION
**Pull request by MrPetovan**
Fixes #10502

### CODE REVIEW

File: src/Module/Contact.php [hunk 1]
```diff
@@ -59,47 +59,45 @@ private static function batchActions()
...
-		$orig_records = DBA::toArray($stmt);
+		$orig_records = Model\Contact::selectToArray(['id', 'uid'], ['id' => $_POST['contact_batch'], 'uid' => [0, local_user()], 'self' => false, 'deleted' => false]);
 
 		$count_actions = 0;
 		foreach ($orig_records as $orig_record) {
-			$contact_id = $orig_record['id'];
-			if (!empty($_POST['contacts_batch_update'])) {
-				self::updateContactFromPoll($contact_id);
+			$cdata = Model\Contact::getPublicAndUserContactID($orig_record['id'], local_user());
+			if(empty($cdata)) {
```
Comment (annando) [inline src/Module/Contact.php]: Can you add a space after the `if`?

File: src/Module/Contact.php [hunk 2]
```diff
@@ -355,54 +358,45 @@ public static function content(array $parameters = [], $update = 0)
 				throw new NotFoundException(DI::l10n()->t('Contact not found'));
 			}
 
-			if ($cmd === 'update' && ($orig_record['uid'] != 0)) {
-				self::updateContactFromPoll($contact_id);
-				DI::baseUrl()->redirect('contact/' . $contact_id);
+			$cdata = Model\Contact::getPublicAndUserContactID($orig_record['id'], local_user());
```
Comment (annando) [inline src/Module/Contact.php:364]: You haven't checked for an empty `$cdata` here. Should or shouldn't we handle this?

File: src/Module/Contact.php [hunk 3]
```diff
@@ -355,54 +358,45 @@ public static function content(array $parameters = [], $update = 0)
...
-				DI::baseUrl()->redirect('contact/' . $contact_id);
+			if ($cmd === 'updateprofile' && $cdata['user']) {
+				self::updateContactFromProbe($cdata['user']);
+				DI::baseUrl()->redirect('contact/' . $cdata['public']);
 				// NOTREACHED
 			}
 
 			if ($cmd === 'block') {
-				if (public_contact() === $contact_id) {
+				if (public_contact() === $cdata['public']) {
```
Comment (annando) [inline src/Module/Contact.php:382]: Shouldn't we add this check to the batch operations as well?

**Review by annando — COMMENTED** [review]
I just saw some few things.

File: src/Module/Contact.php (hunk 3 above)
Comment (MrPetovan) [reply src/Module/Contact.php:382]: We can but it's less important, the batch operation is a POST request which makes it impossible to craft a malicious link to make someone block themselves or drop their own contact.

File: src/Module/Contact.php (hunk 2 above)
Comment (MrPetovan) [reply src/Module/Contact.php:364]: We should check the output of most methods, you're right.

File: src/Module/Contact.php (hunk 3 above)
Comment (annando) [reply src/Module/Contact.php:382]: But we should prevent that someone deletes their own contact by mistake, shouldn't we?

Comment (MrPetovan) [reply src/Module/Contact.php:382]: Yes we should, but it isn't that pressing, we don't generate a POST form where your own contact is featured, and #10672 should prevent someone to craft a malicious cross-posting form.

**Pull request merged**